jerry-ext/handler/handler-assert.c

```diff
@@ -12,36 +12,64 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+#include "jerryscript-core.h"
 #include "jerryscript-ext/handler.h"
+#include "jext-common.h"
 #include "jerryscript-port.h"
 
 #include <inttypes.h>
 
+typedef enum
+{
+  JERRYX_OP_EQUAL = 0,
+  JERRYX_OP_STRICT_EQUAL = 1,
+  JERRYX_OP_NOT_EQUAL = 2,
+  JERRYX_OP_NOT_STRICT_EQUAL = 3
+} jerryx_equality_operation_t;
+
+JERRYX_STATIC_ASSERT ((int) (JERRYX_OP_EQUAL) == (int) (JERRY_BIN_OP_EQUAL),
+                      jerryx_op_equal_must_be_equal_to_jerry_bin_op_equal);
+
+JERRYX_STATIC_ASSERT ((int) (JERRYX_OP_STRICT_EQUAL) == (int) (JERRY_BIN_OP_STRICT_EQUAL),
+                      jerryx_op_strict_equal_must_be_equal_to_jerry_bin_op_strict_equal);
+
 /**
- * Hard assert for scripts. The routine calls jerry_port_fatal on assertion failure.
- *
- * Notes:
- *  * If the `JERRY_FEATURE_LINE_INFO` runtime feature is enabled (build option: `JERRY_LINE_INFO`)
- *    a backtrace is also printed out.
- *
- * @return true - if only one argument was passed and that argument was a boolean true.
- *         Note that the function does not return otherwise.
+ * return an error message if you call this fuction with NULL paramater
+ * or if you call this fuction with a real parameter and that is a string
+ * this function returns with a message and the error message.
  */
-jerry_value_t
-jerryx_handler_assert_fatal (const jerry_value_t func_obj_val, /**< function object */
-                             const jerry_value_t this_p, /**< this arg */
-                             const jerry_value_t args_p[], /**< function arguments */
-                             const jerry_length_t args_cnt) /**< number of function arguments */
+static void
+jerryx_fatal (const jerry_value_t *message)
 {
-  (void) func_obj_val; /* unused */
-  (void) this_p; /* unused */
-
-  if (args_cnt == 1
-      && jerry_value_is_boolean (args_p[0])
-      && jerry_get_boolean_value (args_p[0]))
+  if (message != NULL && jerry_value_is_string (*message))
   {
-    return jerry_create_boolean (true);
+    jerry_length_t msg_length = jerry_get_string_length (*message);
+    jerry_length_t curr_length = msg_length;
```

Comment:
You could just do `jerry_length_t curr_length = jerry_get_string_length (*message);` and remove the `msg_length` variable.